java/src/org/openqa/selenium/grid/node/ForwardWebDriverCommand.java

```diff
@@ -17,19 +17,30 @@
 
 package org.openqa.selenium.grid.node;
 
+import static org.openqa.selenium.remote.HttpSessionId.getSessionId;
+
 import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.remote.SessionId;
 import org.openqa.selenium.remote.http.HttpHandler;
 import org.openqa.selenium.remote.http.HttpRequest;
 import org.openqa.selenium.remote.http.HttpResponse;
+import org.openqa.selenium.remote.http.Routable;
 
-class ForwardWebDriverCommand implements HttpHandler {
+class ForwardWebDriverCommand implements HttpHandler, Routable {
 
   private final Node node;
 
   ForwardWebDriverCommand(Node node) {
     this.node = Require.nonNull("Node", node);
   }
 
+  @Override
+  public boolean matches(HttpRequest req) {
+    return getSessionId(req.getUri())
+        .map(id -> node.isSessionOwner(new SessionId(id)))
+        .orElse(false);
+  }
+
```

Comment:
Is this code reached? In my mind the the `isSessionOwner` must be checked inside the `execute` before calling  `node.executeWebDriverCommand`.

Reply:
You are correct! I updated and tried to check via a unit test. Can you review one more time